### PR TITLE
gRPC: convert gitserver.Search

### PR DESCRIPTION
**Pull request by camdencheek**
This adds a gRPC implementation of the `gitserver.Search` RPC.

Stacked on #47584 

## Test plan

Integration tests. Manual poking and inspecting results/errors. Ran unit tests with the `grpc` flag both enabled and disabled.

### CODE REVIEW

File: internal/gitserver/v1/gitserver.proto [hunk 1]
```diff
@@ -30,3 +33,136 @@ message ExecStatusPayload {
   int32 status_code = 1;
   string stderr = 2;
 }
+
+message SearchRequest {
+  string repo = 1;
+  repeated RevisionSpecifier revisions = 2;
+  int64 limit = 5;
```
Comment (ggilmore) [inline internal/gitserver/v1/gitserver.proto:40]: Where did field number `3` go?

Comment (camdencheek) [reply internal/gitserver/v1/gitserver.proto:40]: Ha, good question. Will update the field numbers

Comment (camdencheek) [reply internal/gitserver/v1/gitserver.proto:40]: And add doc comments while I'm at it...just noticed I missed that

File: internal/gitserver/v1/gitserver.proto [hunk 2]
```diff
@@ -30,3 +33,136 @@ message ExecStatusPayload {
...
+
+// DiffModifiesFile is a predicate that matches if the commit modifies any files
+// that match the given regex pattern.
+message DiffModifiesFile {
+  string expr = 1;
+  bool ignore_case = 2;
+}
+
+// Boolean is a predicate that will either always match or never match
+message Boolean {
```
Comment (ggilmore) [inline internal/gitserver/v1/gitserver.proto:115]: How do you feel about adding a `Predicate` or `Node` predicate to these message types? 

(the message type named `Boolean` threw me for a loop)

Comment (camdencheek) [reply internal/gitserver/v1/gitserver.proto:115]: Ah, yeah, that's fair. Maybe they should just be defined inside the `Node` type?

Comment (ggilmore) [reply internal/gitserver/v1/gitserver.proto:115]: Ah, I forgot that you could do that - that sounds like a good way of namespacing them

Comment (camdencheek) [reply internal/gitserver/v1/gitserver.proto:115]: Okay, so this didn't work out quite like I hoped. Defining the message in the `QueryNode` type generated types named like `QueryNode_Boolean`, which is great. However, `QueryNode` also has a `oneof` field named `Boolean`, so to disambiguate between the two, it names that one `QueryNode_Boolean_`. Having one type named with an underscore and one without is confusing IMO, so I just took your initial suggestion and added `Node` to all the node types.

**Review by ggilmore — APPROVED** [review]

**Pull request merged**